File: sdk/java/src/main/java/io/apibrew/client/ListRecordParams.java
package io.apibrew.client;

import io.apibrew.client.model.Extension;
import lombok.Builder;
import lombok.Getter;
import lombok.Singular;

import java.util.List;
import java.util.Map;

@Getter
@Builder
public class ListRecordParams {

    // you can either use filters or query, not both
    private final Extension.BooleanExpression query;

    // you can either use filters or query, not both
    private final Map<String, String> filters;

    @Singular
    private final List<String> resolveReferences;

    private final long offset;
    private final int limit;
    private final boolean useHistory;
}

File: sdk/nodejs/src/list-record-params.ts
import {BooleanExpression} from "./model";

export interface ListRecordParams {
    query?: BooleanExpression
    filters?: { [key: string]: string }
    resolveReferences?: string[]
    limit?: number
    offset?: number
    useHistory?: boolean
    aggregation?: Aggregation
    sorting?: SortingItem[]
}

export interface Aggregation {
    items: AggregationItem[]
    grouping: GroupingItem[]
}

export interface AggregationItem {
    name: string
    algorithm: AggregationAlgorithm
    property: string
}

export interface GroupingItem {
    property: string
}

export interface SortingItem {
    property: string
    direction: Direction
}

export enum Direction {
    ASC = "ASC",
    DESC = "DESC"
}

export enum AggregationAlgorithm {
    SUM = "SUM",
    AVG = "AVG",
    MIN = "MIN",
    MAX = "MAX",
    COUNT = "COUNT"
}
